internal/pkg/agent/cmd/container_init_linux.go

```diff
@@ -0,0 +1,283 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
```

Comment:
Can you add the explanation and rationale for why this exists from the PR description directly to this file?